cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/JsonLiveViewer/index.tsx

```diff
@@ -65,65 +51,90 @@ const styles = (theme): StyleRules => {
   };
 };
 
-const JsonLiveViewerView: React.FC<ICreateContext & WithStyles<typeof styles>> = (
-  widgetJSONData
-) => {
-  const { classes, pluginName, pluginType, jsonView, setJsonView } = widgetJSONData;
-  const JSONConfig = getJSONConfig(widgetJSONData);
-  const downloadDisabled = pluginName.length === 0 || pluginType.length === 0;
+const DownloadJSONButton = ({ classes, downloadDisabled, onDownloadClick }) => {
+  return (
+    <Tooltip
+      title={
+        downloadDisabled
+          ? 'Download is disabled until the required fields are filled in'
+          : 'Download Plugin JSON'
+      }
+      classes={{
+        tooltip: classes.jsonViewerTooltip,
+      }}
+    >
+      <div>
+        <Button disabled={downloadDisabled} onClick={onDownloadClick}>
+          <GetAppIcon />
+        </Button>
+      </div>
+    </Tooltip>
```

Comment:
this seems to be the same as the tooltip being used in another component. Worth extracting this out

Reply:
This `JSONDownloadButton` is later extracted in refactor PR #12310 

https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/PluginJSONMenu/JSONActionButtons/JSONDownloadButton/index.tsx